gapii/cc/spy.cpp

```diff
@@ -73,102 +107,41 @@ const uint32_t kStartMidExecutionCapture =  0xdeadbeef;
 
 const int32_t  kSuspendIndefinitely = -1;
 
-inline bool isLittleEndian() {
-    union {
-        uint32_t i;
-        char c[4];
-    } u;
-    u.i = 0x01020304;
-    return u.c[0] == 4;
-}
-
 core::Mutex gMutex;  // Guards gSpy.
 std::unique_ptr<gapii::Spy> gSpy;
 
 } // anonymous namespace
 
 namespace gapii {
 
-#if TARGET_OS == GAPID_OS_ANDROID
-
-/**
- * Parses "[key]: [value]" pairs from getprop output and returns them as a map.
- */
-static std::unordered_map<std::string, std::string> parseSystemProperties(FILE *f) {
-  std::unordered_map<std::string, std::string> result;
-  std::string key, val;
-  int m = 0; // Parsing state (m=0)[(m=1)key](m=2):(m=3)whitespace[(m=4)value](m=0)...
-  for (int x = getc(f); x != EOF; x = getc(f)) {
-    switch (m) {
-    case 0:
-        if (x == '[') {
-            key.clear(); m = 1;
-            break;
-        } else if (isspace(x)) {
-            break;
-        }
-        return result; // Unexpected value.
-    case 1:
-        if (x == ']') {
-            m = 2;
-        } else {
-            key.push_back((char) x);
-        }
-        break;
-    case 2:
-        if (x == ':') {
-            m = 3;
-            break;
-        }
-        return result; // Unexpected value.
-    case 3:
-        if (x == '[') {
-            val.clear(); m = 4;
-            break;
-        } else if (isspace(x)) {
-            break;
-        }
-        return result;
-    case 4:
-        if (x != '\n' && x != '\r') {
-            val.push_back((char) x);
-        } else {
-            if (val.back() == ']') {
-              val.pop_back();
+Spy* Spy::get() {
+    bool init;
```

Comment:
I am not sure about the status of support in the NDK, but I think std::call_once might help simplify this.